vesc_msgs/CMakeLists.txt

```diff
@@ -1,20 +1,45 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.8)
 project(vesc_msgs)
 
-find_package(catkin REQUIRED COMPONENTS
-  std_msgs
-  message_generation
+if(CMAKE_COMPILER_IS_GNUCXX
+   OR CMAKE_CXX_COMPILER_ID
+      MATCHES
+      "Clang"
 )
+  add_compile_options(
+    -Wall
+    -Wextra
+    -Wpedantic
+  )
+endif()
 
-add_message_files(
-  DIRECTORY msg
-  FILES VescState.msg VescStateStamped.msg
-)
+find_package(ament_cmake REQUIRED)
+find_package(builtin_interfaces REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+find_package(std_msgs REQUIRED)
 
-generate_messages(
-  DEPENDENCIES std_msgs
-)
+set(msg_files "msg/VescState.msg" "msg/VescStateStamped.msg")
 
-catkin_package(
-  CATKIN_DEPENDS std_msgs message_runtime
+rosidl_generate_interfaces(
+  ${PROJECT_NAME}
+  ${msg_files}
+  DEPENDENCIES
+  builtin_interfaces
```

Comment:
I'm not sure what the reference was, but I removed it because it does not appear to be necessary as far as the latest official documentation is concerned.